sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -690,6 +690,7 @@ class SparkSession private(
    * @since 2.0.0
    */
   def stop(): Unit = {
+    streams.stopAllQueries()
```

Comment:
I did look into that method. It does not work because the [application end event is generated]( https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L1897) after the [SparkContext stopped flag is set]( https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L1888) which prevents the queries from stopping properly. We could add a new event type that gets called before stopped flag is set in the close method. Do you think that would be an appropriate solution?